woodwork/logical_types.py

```diff
@@ -541,7 +541,7 @@ def validate(self, series):
 
     def transform(self, series):
         """Validates the series and converts the dtype to match the logical type's if it is different."""
-        self._validate_data(series)
+        self.validate(series)
```

Comment:
Ahhh, totally missed that this gets used here. Sorry about that!

It's causing errors because of the dtype matching requirement, I assume (`transform` is the thing that makes sure that the dtype matches). I think it might be best to go back to ahving a separate method for validating the ordinal categories. It can be called `_validate_order_values`, so hopefully that'll be a bit less confusing than `_validate_data`.

Reply:
renamed